{"name": ["shgo_slsqp_sobol_cube", "shgo_powell_sobol_cube", "shgo_nelder_sobol_cube", "scipy_slsqp_cube", "scipy_powell_cube", "scipy_lbfgsb_cube", "pysot_ei_cube", "pysot_lcb_cube", "pysot_random_cube", "pysot_srbf_cube", "pysot_dycors_cube", "ax_default_cube", "optuna_cmaes_cube", "optuna_tpe_cube", "optuna_random_cube", "optuna_random_cube_clone", "optuna_random_cube_clone_1", "optuna_random_cube_clone_2", "platypus_genetic_cube", "platypus_evolutionary_cube", "platypus_nsgaii_cube", "platypus_cmaes_cube", "platypus_gde3_cube", "platypus_ibea_cube", "platypus_moead_cube", "platypus_omopso_cube", "platypus_smpso_cube", "platypus_spea2_cube", "platypus_epsmoea_cube", "nevergrad_ngopt_cube", "nevergrad_ngopt4_cube", "nevergrad_de_cube", "nevergrad_portfolio_cube", "nevergrad_oneplus_cube", "nevergrad_hammersley_cube", "nevergrad_ngopt8_cube", "hyperopt_tpe_cube", "hyperopt_atpe_cube", "hyperopt_rand_cube", "pymoo_nelder_cube", "pymoo_nsga3_cube", "pymoo_unsga3_cube", "pymoo_pattern_cube", "pymoo_nsga2_cube", "pymoo_unsga3_cube", "skopt_gp_default_cube", "bayesopt_ucb_cube", "bayesopt_ei_cube", "bayesopt_poi_cube", "nlopt_direct_cube", "nlopt_ags_cube", "nlopt_esch_cube", "nlopt_isres_cube", "nlopt_crs2lm_cube", "nlopt_directr_cube", "nlopt_directo_cube", "nlopt_directl_cube", "bobyqa_default_cube", "bobyqa_noise_cube", "ultraopt_random_cube", "ultraopt_forest_cube", "ultraopt_gbrt_cube", "ultraopt_etpe_cube", "dlib_cube", "dlib_default_cube", "hebo_cube_sequential", "hebo_cube_batch", "hebo_sequential_cube", "hebo_batch_cube"], "count": [75, 61, 74, 139, 36, 29, 201, 214, 217, 232, 177, 203, 198, 210, 190, 222, 214, 229, 204, 217, 199, 232, 214, 189, 223, 198, 205, 214, 230, 213, 217, 229, 214, 190, 208, 213, 194, 197, 219, 198, 225, 455, 215, 199, 0, 203, 206, 224, 202, 195, 209, 224, 203, 193, 189, 197, 219, 217, 226, 71, 54, 57, 61, 53, 58, 37, 48, 0, 14], "rating": [1304.1051107054973, 1305.1418871567353, 1240.4895959014411, 1681.4804594493949, 1308.0145078870514, 1546.2233636202748, 2117.155121580359, 2031.1450705899406, 1064.5302866053678, 2530.104231274638, 2426.4768318214697, 2393.9023174414374, 1977.434198287909, 2064.7336099188974, 1600, 1600, 1600, 1600, 983.4744386265726, 963.202837071718, 1031.1653479197976, 843.5250100488197, 938.7546157707768, 1108.575503721182, 1419.7990286063155, 1526.815481432943, 1252.5065231059618, 757.5906971101994, 1331.9260551801988, 1922.6159415932736, 1976.6548120692864, 1527.6520426061, 1284.8669227368339, 2378.225979071371, 768.4895274629096, 1775.7029471807577, 1587.786744658013, 1647.7893314369942, 1563.452025713771, 2047.446563311378, 2142.1280437838095, 2016.7878613779258, 2195.825824821768, 1030.0723817206178, 1600, 2314.4032996670335, 2341.0172307960947, 2373.4291039300488, 2365.8648574729445, 2425.2351554891925, 2116.9626312551836, 1530.0823355679288, 1089.33065590968, 1357.3224721241115, 2430.053812629384, 1499.9389336151585, 2605.6768138216, 2304.483966515772, 2409.818353468967, 1317.3157756366866, 1918.1146445155698, 1835.5943713726092, 1941.257240116273, 1886.8480343268375, 1956.145655953539, 1925.1161229648428, 1856.4903955959294, 1600, 1815.4267263738975], "traceback": ["passing", "passing", "passing", "passing", "Traceback (most recent call last):\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/comparison/eloratings.py\", line 76, in optimizer_game\n    black_best_val, black_best_x, black_feval_count = black(objective, n_trials=n_black_trials,\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/optimizers/scipycube.py\", line 40, in scipy_powell_cube\n    return scipy_cube(objective=objective, n_trials=n_trials, n_dim=n_dim, with_count=with_count, method='powell')\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/optimizers/scipycube.py\", line 29, in scipy_cube\n    result = minimize(_objective, x0=[0]*n_dim, method=options['method'],bounds=bounds, options=options)\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/scipy/optimize/_minimize.py\", line 614, in minimize\n    return _minimize_powell(fun, x0, args, callback, bounds, **options)\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/scipy/optimize/optimize.py\", line 3059, in _minimize_powell\n    fx2 = squeeze(func(x2))\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/scipy/optimize/optimize.py\", line 464, in function_wrapper\n    return function(np.copy(x), *(wrapper_args + args))\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/optimizers/scipycube.py\", line 27, in _objective\n    return objective(list(x))\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/objectives/portfolio.py\", line 174, in markowitz_return_on_cube\n    return realized_something_factory(g=g,u=u, adjust=True)\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/objectives/portfolio.py\", line 121, in realized_something_factory\n    w = cube_to_weights(u)\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/objectives/portfolio.py\", line 108, in cube_to_weights\n    raise ValueError(\"u should be in hypercube\")\nValueError: u should be in hypercube\n", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "Black took 211 function evals when instructed to use 211", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "Black took 211 function evals when instructed to use 211", "passing", "passing", "passing", "passing", "passing", "passing", "not yet run", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "Traceback (most recent call last):\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/hebo/acq_optimizers/evolution_optimizer.py\", line 119, in optimize\n    res   = minimize(prob, algo, ('n_gen', self.iter), verbose = self.verbose)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/optimize.py\", line 85, in minimize\n    res = algorithm.solve()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 226, in solve\n    self._solve(self.problem)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 321, in _solve\n    self.next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 246, in next\n    self._next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/algorithms/genetic_algorithm.py\", line 93, in _next\n    self.off = self.mating.do(self.problem, self.pop, self.n_offsprings, algorithm=self)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/infill.py\", line 40, in do\n    _off = self.eliminate_duplicates.do(_off, pop, off)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 26, in do\n    pop = pop[~self._do(pop, None, np.full(len(pop), False))]\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 75, in _do\n    D = self.calc_dist(pop, other)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 66, in calc_dist\n    D = cdist(X, X)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/util/misc.py\", line 90, in cdist\n    return scipy.spatial.distance.cdist(A, B, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/scipy/spatial/distance.py\", line 2954, in cdist\n    return cdist_fn(XA, XB, out=out, **kwargs)\nValueError: Unsupported dtype object\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/comparison/eloratings.py\", line 76, in optimizer_game\n    black_best_val, black_best_x, black_feval_count = black(objective, n_trials=n_black_trials,\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/optimizers/hebocube.py\", line 52, in hebo_sequential_cube\n    return hebo_cube_factory(objective=objective, n_trials=n_trials, n_dim=n_dim, with_count=with_count, n_suggestions=1)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/optimizers/hebocube.py\", line 39, in hebo_cube_factory\n    rec = opt.suggest(n_suggestions=n_suggestions) # <-- don't change this\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/hebo/optimizers/hebo.py\", line 124, in suggest\n    rec = opt.optimize(initial_suggest = best_x, fix_input = fix_input).drop_duplicates()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/hebo/acq_optimizers/evolution_optimizer.py\", line 125, in optimize\n    res   = minimize(prob, algo, ('n_gen', self.iter), verbose = self.verbose)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/optimize.py\", line 85, in minimize\n    res = algorithm.solve()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 226, in solve\n    self._solve(self.problem)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 321, in _solve\n    self.next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 246, in next\n    self._next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/algorithms/genetic_algorithm.py\", line 93, in _next\n    self.off = self.mating.do(self.problem, self.pop, self.n_offsprings, algorithm=self)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/infill.py\", line 40, in do\n    _off = self.eliminate_duplicates.do(_off, pop, off)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 26, in do\n    pop = pop[~self._do(pop, None, np.full(len(pop), False))]\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 75, in _do\n    D = self.calc_dist(pop, other)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 66, in calc_dist\n    D = cdist(X, X)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/util/misc.py\", line 90, in cdist\n    return scipy.spatial.distance.cdist(A, B, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/scipy/spatial/distance.py\", line 2954, in cdist\n    return cdist_fn(XA, XB, out=out, **kwargs)\nValueError: Unsupported dtype object\n", "passing"], "active": [true, true, true, true, true, true, true, true, true, true, true, false, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, false, false, false, false]}